### PR TITLE
Openshift4 Migration.

### PR DESCRIPTION
**Pull request by pwei1018**
*Issue #:* /bcgov/entity###

*Description of changes:*
1. Old templates and Jenkins file clean up;
2. Make file update;
3. CI/CD flow update;

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: data-reset-tool/k8s/templates/bc.yaml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+        # Create working directory
+        RUN mkdir /opt/app-root && chmod 755 /opt/app-root
+        WORKDIR /opt/app-root
+
+        # Install the requirements
+        COPY ./requirements.txt .
+
+        #RUN pip install --upgrade pip
+        RUN pip install pip==20.1.1
```
Comment (severinbeauvais) [inline data-reset-tool/k8s/templates/bc.yaml:50]: This is different from the other BC files. Is it OK?

File: jobs/update-legal-filings/k8s/templates/bc.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+        # Create working directory
+        RUN mkdir /opt/app-root && chmod 755 /opt/app-root
+        WORKDIR /opt/app-root
+
+        # Install the requirements
+        COPY ./requirements.txt .
+
+        #RUN pip install --upgrade pip
+        RUN pip install pip==20.1.1
```
Comment (severinbeauvais) [inline jobs/update-legal-filings/k8s/templates/bc.yaml:50]: Again, is special PIP version required?

File: colin-api/k8s/templates/bc.yaml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        RUN mkdir /opt/app-root && chmod 755 /opt/app-root
+        WORKDIR /opt/app-root
+
+        # Install the requirements
+        COPY ./requirements.txt .
+
+        RUN pip install --upgrade pip
+        RUN pip install --no-cache-dir -r requirements.txt
+
+        COPY . ./src
```
Comment (severinbeauvais) [inline colin-api/k8s/templates/bc.yaml:66]: The other BC files did `COPY . .`. Is this difference here OK?

**Review by severinbeauvais — APPROVED** [review]

File: jobs/update-legal-filings/k8s/templates/bc.yaml (hunk 2 above)
Comment (pwei1018) [reply jobs/update-legal-filings/k8s/templates/bc.yaml:50]: There were issue to use new version pip to build the application. Need someone to verify it we can use other version of pip.

File: data-reset-tool/k8s/templates/bc.yaml (hunk 1 above)
Comment (pwei1018) [reply data-reset-tool/k8s/templates/bc.yaml:50]: See the following comment.

File: devops/openshift/templates/nsp.yaml [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        app: ${APP_NAME}
+    spec:
+      # This name will be used internally by Aporeto; it should match
+      # the `name` field in metadata above.
+      description: |
+        specify a custom external network that can be
+        referenced by name. This allows pods to talk out to the
+        internet as needed.
+      entries:
+        - dev.oidc.gov.bc.ca
```
[the remark this answers is not in the document]
Comment (pwei1018) [reply devops/openshift/templates/nsp.yaml:62]: This file should be delete. We don't use nsp anymore.

File: jobs/update-legal-filings/k8s/templates/bc.yaml (hunk 2 above)
Comment (severinbeauvais) [reply jobs/update-legal-filings/k8s/templates/bc.yaml:50]: If the current script works then this conversation can be resolved, but we should have a tech debt ticket to verify/upgrade the pip version.

File: data-reset-tool/k8s/templates/bc.yaml (hunk 1 above)
Comment (severinbeauvais) [reply data-reset-tool/k8s/templates/bc.yaml:50]: If the current script works then this conversation can be resolved, but we should have a tech debt ticket to verify/upgrade the pip version.

**Pull request merged**